src/scheduler/listDagRuns.tsx

```diff
@@ -0,0 +1,242 @@
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    page
+  } = useTable(
+    //@ts-ignore react-table 'columns' which is declared here on type 'TableOptions<ICluster>'
+    { columns, data, autoResetPage: false, initialState: { pageSize: 50 } },
+    useGlobalFilter,
+    usePagination
+  );
+
+  const tableDataCondition = (cell: ICellProps) => {
+    if (cell.column.Header === 'State') {
+      if (cell.value === 'success') {
+        return (
+          <div className="dag-run-state-parent">
+            <td
+              {...cell.getCellProps()}
+              className="dag-runs-table-data-state-success"
+            >
+              {cell.render('Cell')}
+            </td>
+          </div>
+        );
+      } else if (cell.value === 'failed') {
+        return (
+          <div className="dag-run-state-parent">
+            <td
+              {...cell.getCellProps()}
+              className="dag-runs-table-data-state-failure"
+            >
+              {cell.render('Cell')}
+            </td>
+          </div>
+        );
+      } else if (cell.value === 'running') {
+        return (
+          <div className="dag-run-state-parent">
+            <td
+              {...cell.getCellProps()}
+              className="dag-runs-table-data-state-running"
+            >
+              {cell.render('Cell')}
+            </td>
+          </div>
+        );
+      } else if (cell.value === 'queued') {
+        return (
+          <div className="dag-run-state-parent">
+            <td
+              {...cell.getCellProps()}
+              className="dag-runs-table-data-state-queued"
+            >
+              {cell.render('Cell')}
+            </td>
+          </div>
+        );
+      }
+    }
+    return (
+      <td {...cell.getCellProps()} className="notebook-template-table-data">
+        {cell.render('Cell')}
+      </td>
+    );
+  };
+
+  const listDagRunsList = async () => {
+    await SchedulerService.listDagRunsListService(
+      composerName,
+      dagId,
+      startDate,
+      endDate,
+      setDagRunsList,
+      setDagRunId,
+      setIsLoading,
+      setBlueListDates,
+      setGreyListDates,
+      setOrangeListDates,
+      setRedListDates,
+      setGreenListDates,
+      setDarkGreenListDates
+    );
+  };
+
+  useEffect(() => {
+    listDagRunsList();
+  }, [startDate, endDate]);
+
+  useEffect(() => {
+    if (selectedDate !== null && dagRunsList.length > 0) {
+      let currentDate = new Date(selectedDate.toDate()).toDateString();
+      let currentDateDagRunList: any = dagRunsList.filter((dagRun: any) => {
+        return dagRun.date === currentDate;
+      });
+      if (currentDateDagRunList.length > 0) {
+        setDagRunsCurrentDateList(currentDateDagRunList);
+        setDagRunId(
+          currentDateDagRunList[currentDateDagRunList.length - 1].dagRunId
+        );
+      } else {
+        setDagRunsCurrentDateList([]);
+        setDagRunId('');
```

Comment:
Since this is duplicated on line 185, we should change the conditional flow to avoid having this else block again.

Reply:
Resolved - Changed the conditional flow